covid19-etl/etl/idph.py

```diff
@@ -66,169 +65,205 @@ def files_to_submissions(self):
-                    date, county, demographic
+                summary_location, summary_clinical = self.parse_county_data_for_date(
+                    date_str, county, data
                 )
-
-                self.summary_locations.append(summary_location)
+                summary_clinical = {
+                    **summary_clinical,
+                    **demographics[date_str],  # add demographics data
+                }
+
+                self.summary_locations[
+                    summary_location["submitter_id"]
+                ] = summary_location
                 self.summary_clinicals.append(summary_clinical)
 
-            for illinois_data in data["state_testing_results"]["values"]:
-                illinois_historic_data = self.parse_historical_data(illinois_data)
-                self.summary_clinicals.append(illinois_historic_data)
-
-    def parse_historical_data(self, illinois_data):
+    def parse_state_data(self, latest_submitted_date):
         """
         Parses historical state-level data. "summary_location" node is created
         from "characteristics_by_county" data.
 
         Args:
-            illinois_data (dict): data JSON with "testDate", "total_tested",
-                "confirmed_cases" and "deaths"
-
-        Returns:
-            dict: "summary_clinical" node for Sheepdog
+            latest_submitted_date (datetime): date for latest submitted date
         """
+        url = "https://idph.illinois.gov/DPHPublicInformation/api/COVIDExport/GetIllinoisCases"
+        print("Getting state data from {}".format(url))
         county = "Illinois"
 
-        date = datetime.datetime.strptime(
-            illinois_data["testDate"], "%m/%d/%Y"
-        ).strftime("%Y-%m-%d")
-
-        (
-            summary_location_submitter_id,
-            summary_clinical_submitter_id,
-        ) = self.get_location_and_clinical_submitter_id(county, date)
+        demographics = self.get_demographics(
+            # we need to start the day after `latest_submitted_date` but this is easier
+            start_date=latest_submitted_date,
+            end_date=datetime.date.today(),
+            county=county,
+        )
 
-        summary_clinical = {
-            "submitter_id": summary_clinical_submitter_id,
-            "date": date,
-            "confirmed": illinois_data["confirmed_cases"],
-            "testing": illinois_data["total_tested"],
-            "deaths": illinois_data["deaths"],
-            "summary_locations": [{"submitter_id": summary_location_submitter_id}],
-        }
+        with closing(self.get(url, stream=True)) as r:
+            daily_data = r.json()
+            for illinois_data in daily_data:
+                date = datetime.datetime.strptime(
+                    illinois_data["testDate"], "%Y-%m-%dT%H:%M:%S"
+                )
+                date_str = date.strftime("%Y-%m-%d")
+                if not latest_submitted_date or date <= latest_submitted_date:
```

Comment:
Raising it here as well. If previous code snippet was intentional we can resolve this one as well.